### PR TITLE
Kevin Rodgers Code Review 2

### PR DESCRIPTION
**Pull request by Rodgerskc**
See review for comments.

### CODE REVIEW

File: src/main/java/com/neighborhoodAssurance/NAController.java [hunk 1]
```diff
@@ -3,19 +3,25 @@
...
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.converter.xml.SourceHttpMessageConverter;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.servlet.ModelAndView;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.RequestParam;
```
Comment (Rodgerskc) [inline src/main/java/com/neighborhoodAssurance/NAController.java:14]: Logging imports.

File: src/main/java/com/neighborhoodAssurance/NAController.java [hunk 2]
```diff
@@ -3,19 +3,25 @@
...
 
 @Controller
 public class NAController {
-	
+
+	Logger logger = LoggerFactory.getLogger(NAController.class);
+
```
Comment (Rodgerskc) [inline src/main/java/com/neighborhoodAssurance/NAController.java:20]: I added logging for you, please feel free to use this template + imports throughout your project to debug your issues.

File: src/main/java/com/neighborhoodAssurance/NAController.java [hunk 3]
```diff
@@ -3,19 +3,25 @@
...
 	@Autowired
 	private IAgencyService agencyServiceStub;
 
-	@RequestMapping(value="/home", method=RequestMethod.GET) 
+	@RequestMapping(value="/", method=RequestMethod.GET)
```
Comment (Rodgerskc) [inline src/main/java/com/neighborhoodAssurance/NAController.java:24]: I fixed your base routing, when I first started your project it didn't work.

File: src/main/java/com/neighborhoodAssurance/NAController.java [hunk 4]
```diff
@@ -28,4 +34,10 @@ public ModelAndView home() {
 	public String readJSON() {
 		return "home";
 	}
+
+	@RequestMapping(value="/searchResults", method=RequestMethod.GET)
+	public String doSearch(@RequestParam String searchTerm, @RequestParam int submit) {
+		logger.info("Request successful we got: " + searchTerm);
+		return "home";
+	}
 }
```
Comment (Rodgerskc) [inline src/main/java/com/neighborhoodAssurance/NAController.java:43]: Once I got your site building appropriately I noticed that you commented out a form. I removed the comments and hooked it up so it works. Currently it only logs out the message but you get the idea. Feel free to use the searchTerm param however you see fit in your searching function. You can obviously expand on this quite a bit.

File: src/main/resources/static/stylesheet.css [hunk 5]
```diff
@@ -2,5 +2,5 @@
 
 h1 {
 	text-align: center;
-	color: light-blue;
+	color: deepskyblue;
```
Comment (Rodgerskc) [inline src/main/resources/static/stylesheet.css:5]: I wasn't able to find light-blue as a qualified color name, so changed it to deepskyblue. Here's a list of color names if you're not interested in using the hex values. https://www.rapidtables.com/web/css/css-color.html

File: src/main/resources/templates/home.html [hunk 6]
```diff
@@ -4,7 +4,7 @@
 <head>
 	<meta charset="ISO-8859-1"/>
 	<link rel="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css">
-	<link rel="stylesheet" type="text/css" href="../static/stylesheet.css" />
+	<link rel="stylesheet" type="text/css" href="stylesheet.css" />
```
Comment (Rodgerskc) [inline src/main/resources/templates/home.html:7]: Your stylesheet path didn't resolve, if you see issues with your page open your developer panel in the browser. In most browsers the keybind is ctrl + shift + i. If you hit the network tab and refresh the page (ctrl + r) you will see the requests that the page made while loading. This is incredibly useful for debugging problems with your API's or even with stuff like this. Once I saw that your path wasn't working, it was pretty easy to fix.

File: src/main/resources/templates/home.html [hunk 7]
```diff
@@ -20,11 +20,11 @@ <h3>This program allows users to insert an address and return criminal statistic
 	Your agency is:
 	<p th:text="${agencyDTO}"/>
 	
-	<!--Search to find out about your desired location...
+	<!--Search to find out about your desired location...-->
 	<form action="#" th:action="@{/searchResults}" method="GET">
 		<input type="text" name="searchTerm" placeholder="Test: Search for OH"/>
 		<button name="submit" type="submit" value="0">Submit</button>
-	</form>-->
+	</form>
```
Comment (Rodgerskc) [inline src/main/resources/templates/home.html:27]: This form accurately routes to the /searchResults endpoint I created for you.

**Review by Rodgerskc — COMMENTED** [review]
**Analysis of the program:**
You guys still have a ways to go. It looks like you've got some pieces in the right places but you need to flesh out some of the core functionality of your site. I fixed a few of your problems, see comments for more info.
**Was the program available in UC Github on time?**
Yes.
**Is the program documented/commented well enough for you to understand?**
There's not much in the way of comments but your readme does explain the concept of your project well.
**Does the program compile?**
Technically yes, although there were several issues out of the box such as the homepage not displaying, and the stylesheet link being to the wrong path.

**Overall**
I think you guys have a great start and there's plenty of time left in the semester to get your project into a good place by the end. I believe you guys can do it!

File: src/main/java/com/neighborhoodAssurance/NAController.java (hunk 2 above)
Comment (discospiff) [reply src/main/java/com/neighborhoodAssurance/NAController.java:20]: This is a good thing to identify/add in a code review.  Are the logging levels appropriate?

File: src/main/java/com/neighborhoodAssurance/NAController.java (hunk 3 above)
Comment (discospiff) [reply src/main/java/com/neighborhoodAssurance/NAController.java:24]: If a default endpoint does not exist... good idea to add one.

File: src/main/resources/static/stylesheet.css (hunk 5 above)
Comment (discospiff) [reply src/main/resources/static/stylesheet.css:5]: Good catch.

File: src/main/resources/templates/home.html (hunk 6 above)
Comment (discospiff) [reply src/main/resources/templates/home.html:7]: Very helpful tip!

**Comment by discospiff** [comment]
An outstanding, thorough review.  The reviewer did an excellent job making suggestions, decreasing the technical debt of this program.  Merging these changes will add significant value to this project.

**Pull request merged**